embedded/store/key_reader.go

```diff
@@ -313,16 +335,26 @@ func (v *valueRef) Len() uint32 {
 	return v.valLen
 }
 
-func (r *KeyReader) ReadBetween(initialTxID, finalTxID uint64) (key []byte, val ValueRef, tx uint64, err error) {
+type storeKeyReader struct {
+	snap           *Snapshot
+	reader         *tbtree.Reader
+	filters        []FilterFn
+	refInterceptor valueRefInterceptor
+
+	offset  uint64
+	skipped uint64
+}
+
+func (r *storeKeyReader) ReadBetween(initialTxID, finalTxID uint64) (key []byte, val ValueRef, err error) {
```

Comment:
it would be good to add comments for functions so it is easy to understand what they do. I.e. what ReadBetween function does, what Read function does etc.